### DISCUSSION TITLE
Is there any way to focus on an SWT Push type tool item directly and not on the parent tool bar?

### DISCUSSION THREAD

**Discussion by purnac123** [eclipse-platform/eclipse.platform.swt#164] [Q&A]
Is there any way to focus on an SWT Push type tool item directly and not on the parent tool bar?
I would also like to know if I can send accessible change listeners to tool items directly in case I wanted screen readers to read out messages on selection of a tool item.

**Comment by merks** [comment]
It seems to me that the first tool button is given focus (when using Tab to navigate) and after that the arrow keys move the 'focus" between the buttons.  Hitting space invokes the tool button action so I think it has focus...

**Comment by purnac123** [comment]
Hi Ed,

Thank you so much for your reply.
For my scenario I wanted know whether I can alter the default behavior and set focus on a particular tool item instead of the default focus. For example, say on selecting the first tool item using space bar, I want to move focus to the second one using setfocus or forcefocus. I have seen this work in CHECK and RADIO type tool items using the Select action but am unable to find something similar for PUSH type tool items.

**Comment by merks** [comment]
I see.  Someone with more expertise than me will need to have a look at your question...

**Comment by lshanmug** [comment]
> Hi Ed,
> 
> Thank you so much for your reply. For my scenario I wanted know whether I can alter the default behavior and set focus on a particular tool item instead of the default focus. For example, say on selecting the first tool item using space bar, I want to move focus to the second one using setfocus or forcefocus. 

There is no API to directly set focus on a particular ToolItem.

> I have seen this work in CHECK and RADIO type tool items using the Select action but am unable to find something similar for PUSH type tool items.

If the referred UI/code is in Eclipse, can you please point to it?

**Reply by purnac123** [reply]
Hi @lshanmug 

Thank you for your reply.

The toolitem class is as mentioned below:

https://github.com/eclipse-platform/eclipse.platform.swt/blob/master/bundles/org.eclipse.swt/Eclipse%20SWT/gtk/org/eclipse/swt/widgets/ToolItem.java.

The ToolBar is below:

https://github.com/eclipse-platform/eclipse.platform.swt/blob/master/bundles/org.eclipse.swt/Eclipse%20SWT/gtk/org/eclipse/swt/widgets/ToolBar.java

If I perform a setFocus or forceFocus on a ToolBar by default the first ToolItem of the ToolBar gets selected.
My requirement is to select a particular ToolItem instead of always selecting the first.
